hawc/apps/summary/table_serializers.py

```diff
@@ -0,0 +1,304 @@
+
+import pandas as pd
+from rest_framework import serializers
+
+from ..animal import models as animal_models
+from ..assessment.models import Assessment
+from ..common.serializers import FlexibleFieldsMixin
+from ..materialized.models import FinalRiskOfBiasScore
+from ..study.models import Study
+from . import constants
+
+
+class SummaryTableDataSerializer(serializers.Serializer):
+    table_type = serializers.ChoiceField(constants.TableType.choices)
+
+    @property
+    def cache_key(self):
+        return (
+            f"table-type-{self.validated_data['table_type']}-{self.validated_data['ser'].cache_key}"
+        )
+
+    def validate(self, data):
+        if data["table_type"] == constants.TableType.STUDY_EVALUATION:
+            ser = StudyEvaluationSerializer(data=self.initial_data, context=self.context)
+            ser.is_valid(raise_exception=True)
+            data["ser"] = ser
+        return data
+
+    def get_data(self):
+        ser = self.validated_data.get("ser")
+        if ser is not None:
+            return ser.get_data()
+
+
+class StudySerializer(FlexibleFieldsMixin, serializers.ModelSerializer):
+    class Meta:
+        model = Study
+        fields = ["id", "short_citation"]
+        read_only_fields = fields
+
+
+class ExperimentSerializer(FlexibleFieldsMixin, serializers.ModelSerializer):
+    class Meta:
+        model = animal_models.Experiment
+        fields = ["id", "study_id", "name", "chemical"]
+        read_only_fields = fields
+
+
+class AnimalGroupSerializer(FlexibleFieldsMixin, serializers.ModelSerializer):
+    description = serializers.SerializerMethodField()
+    route_of_exposure = serializers.SerializerMethodField()
+    treatment_period = serializers.SerializerMethodField()
+    dose_dict = serializers.SerializerMethodField()
+
+    class Meta:
+        model = animal_models.AnimalGroup
+        fields = [
+            "id",
+            "experiment_id",
+            "name",
+            "description",
+            "route_of_exposure",
+            "treatment_period",
+            "dose_dict",
+        ]
+        read_only_fields = fields
+
+    def get_description(self, obj):
+        return f"{obj.generation}{' ' if obj.generation else ''}{obj.species.name}, {obj.strain.name} ({obj.sex_symbol})"
+
+    def get_route_of_exposure(self, obj):
+        return obj.dosing_regime.get_route_of_exposure_display()
+
+    def get_treatment_period(self, obj):
+        text = obj.experiment.get_type_display()
+        if "(" in text:
+            text = text[: text.find("(")]
+        if obj.dosing_regime.duration_exposure_text:
+            text = f"{text} ({obj.dosing_regime.duration_exposure_text})"
+        return text
+
+    def get_dose_dict(self, obj):
+        dose_dict = OrderedDict()
+        dgs = obj.dosing_regime.doses.all()
+        for dg in dgs:
+            if dg.dose_units.name not in dose_dict:
+                dose_dict[dg.dose_units.name] = str(dg.dose)
```

Comment:
this is annoying, but can we do an `int_or_float()` type thing here; so that values like 0 are shown as 0.0 instead of 0